src/orchard/classpath.clj

```diff
@@ -88,6 +89,28 @@
 
 ;;; Boot's hack - previously part of cider-nrepl
 
+(defn boot-classloader
+  "Creates a class-loader that knows original source files paths in Boot project."
+  []
+  (let [class-path (System/getProperty "fake.class.path")
+        dir-separator (System/getProperty "file.separator")
+        paths (str/split class-path (re-pattern (System/getProperty "path.separator")))
+        urls (map
+              (fn [path]
+                (let [url (if (re-find #".jar$" path)
+                            (str "file:" path)
+                            (str "file:" path dir-separator))]
+                  (new java.net.URL url)))
+              paths)]
+    ;; TODO: Figure out how to add the JDK sources here
+    (new java.net.URLClassLoader (into-array java.net.URL urls))))
+
+(defn boot-hack-classloader
```

Comment:
The name here is a bit misleading, as this might be a boot-classloader or a regular one depending on the project type.

Reply:
Right, `project-based-classloader`?

You can also see that we have similar stuff at the top of the same file, but I didn't dare replacing it - you are right that is not conditional...

Reply:
Already turned off the today, preparing for sleeping, will change tomorrow

Reply:
Maybe `boot-aware-classloader` or `resource-classloader`?

Reply:
I like the former